### DISCUSSION TITLE
Lock File Generation feature flag

### DISCUSSION THREAD

**Discussion by setchy** [renovatebot/renovate#25299] [Suggest an Idea]
### Tell us more.

I'd like to propose a new flag, `lockFileGenerate`, which is defaulted to false.  When set to true it would raise a PR containing a newly generated lock file for managers that support them and when one does not exist in the git already.

This would help orgs/repositories who have a policy of maintaining lockfiles within git, and helps improve the effectiveness of other downstream tooling such as SBOM generators/analysis.

This feature suggestion is similar to the [lockFileRegenerate idea discussed in this comment](https://github.com/renovatebot/renovate/issues/13470#issuecomment-1331864004)

**Comment by rarkins** [comment]
Logically speaking is this feature closer to lockFileMaintenance or more like pinDigests? Code-wide it would probably share a lot with maintenance but in other ways it's like pinning because it only happens once and not on a schedule

**Reply by setchy** [reply]
50/50 i reckon 😄

**Comment by 0963567339** [comment]
[]()
